endpoints/v2/manifest.py

```diff
@@ -399,6 +404,16 @@ def _write_manifest(namespace_name, repo_name, tag_name, manifest_impl):
     if repository_ref is None:
         raise NameUnknown()
 
+    if app.config.get("FEATURE_QUOTA_MANAGEMENT", False):
+        quota = quota_registry_model.verify_namespace_quota_force_cache(
+            namespace_name, repository_ref
+        )
+        if quota["severity_level"] == 1:
+            quota_registry_model.notify_organization_admins("quota_warning", namespace_name)
+        elif quota["severity_level"] == 2:
+            quota_registry_model.notify_organization_admins("quota_error", namespace_name)
+            raise QuotaExceeded()
```

Comment:
Ditto. This looks like the same check as the other endpoint.  Can probably factor this out into a single decorator for all endpoints.